packages/node-sdk/README.md

```diff
@@ -0,0 +1,223 @@
+# Bucket Node.js SDK
+
+Node.js, JavaScriptS/Typescript feature flag and tracking client for [Bucket.co](https://bucket.co).
+
+## Installation
+
+Install using `yarn` or `npm` with:
+
+> `yarn add -s @bucketco/node-sdk` or `npm install -s @bucketco/node-sdk`.
+
+Other languages than JavaScript/Typescript are currently not supported by an SDK.
+You can [use the HTTP API directly](https://docs.bucket.co/reference/http-tracking-api)
+
+## Basic usage
+
+Before the library can be used, you need to obtain a secret key from
+[Environment setting view](https://app.bucket.co/envs/{environment}/settings/app-environments)
+in **Bucket.co**.
+
+> [!CAUTION]
+> Secret keys should only be used by the `node-sdk` and not any other
+> SDKs provided by Bucket. Secret keys offer the users the ability to obtain
+> information that is often sensitive and thus should not be used in any
+> front-end application.
+
+```ts
+import { BucketClient } from "@bucketco/node-sdk";
+
+// Create a new instance of the client with the secret key. Additional options
+// are available, such as supplying a logging sink, a custom HTTP client and
+// other custom properties.
+//
+// We recommend that only one global instance of `client` should be created
+// to avoid multiple round-trips to our servers.
+const client = new BucketClient({
+  secretKey: "sec_prod_xxxxxxxxxxxxxxxxxxxxx",
+});
+
+// Initialize the client's feature flag retrieval loop. You must call this method
+// prior to any calls to `getFlags()`, otherwise an empty object will be returned.
+//
+// You can also supply your `fallbackFlags` to the `initialize()` method. These
+// fallback flags are used in the situation when the server-side flags are not
+// yet loaded or there are issues retrieving them.
+await client.initialize();
+```
+
+At any point where the client needs to be used, we can configure it through
+`withUser()`, `withCompany()` and `withCompanyContext()` methods. Each of
+these three methods returns a new instance of `Client` class which includes
+the applied user, company or context:
+
+```ts
+// configure the client
+const boundClient = client
+  .withUser("john_doe", { attributes: { name: "John Doe" } })
+  .withCompany("acme_inc", { attributes: { name "Acme Inc."} })
+  .withCustomContext({ additional: "context", number: [1,2,3] })
+
+const fallbackFlags = { can_see_new_reports: { value: true } }
+
+// actively wait for the flags to be loaded
+await boundClient.initialize(fallbackFlags)
```

Comment:
Hm, I think fallbackFlags should be an argument to the `Client` constructor instead of to `initialize`?  Feels like it fits into the `Client` constructor with the other initialization arguments.

Reply:
i think the order here is a bit funky. We just showed how to bind the client and here comes the initialization example to show how to use the fallbackFlags. Let's make `fallbackFlags` part of the Client constructor and remove this example. I want the getting started examples to be as simple as possible so the `Client` creation example you have above that just takes the secretKey is good. Let's just explain the fallbackFlag in "Initialization Options" below.

Reply:
Hm, so we won't be able to switch over to new fallback flags when we want? We would only be able to create a new instance of the client when we need new fallback flags.

Reply:
Was the idea that you can call initialize again if you want to change the fallback flags? I’d expect you’d be unable to call initialize more than once so that would be a surprising recommendation to me. I think it’s fine to require folks to create a new client if they want to switch the fallback flags.